modules/caddyhttp/rewrite/rewrite.go

```diff
@@ -194,16 +194,13 @@ func (rewr Rewrite) Rewrite(r *http.Request, repl *caddy.Replacer) bool {
 
 		// before continuing, we need to check if a query string
 		// snuck into the path component during replacements
-		if quPos := strings.Index(newPath, "?"); quPos > -1 {
-			// recompute; new path contains a query string
+		if before, after, found := strings.Cut(newPath, "?"); found {
 			var injectedQuery string
-			newPath, injectedQuery = newPath[:quPos], newPath[quPos+1:]
-			// don't overwrite explicitly-configured query string
+			newPath, injectedQuery = before, after
 			if query == "" {
 				query = injectedQuery
 			}
 		}
-
```

Comment:
This empty line should probably remain since the comment above doesn't apply to these lines